pkg/controller/controller.go

```diff
@@ -84,17 +91,26 @@ func NewManagerForRestConfig(conf *config.Config, rc *rest.Config) (ctrl.Manager
 
 	m.AddHealthzCheck("liveness", func(req *http.Request) error { return nil })
 
-	kcs, err := kubernetes.NewForConfig(rc) // need non-caching client since manager hasn't started yet
+	// create non-caching clients, non-caching for use before manager has started
+	kcs, err := kubernetes.NewForConfig(rc)
+	if err != nil {
+		return nil, err
+	}
+	cl, err := client.New(rc, client.Options{Scheme: scheme})
 	if err != nil {
 		return nil, err
 	}
 
-	log := m.GetLogger()
-	deploy, err := getSelfDeploy(kcs, conf, log)
+	setupLog := m.GetLogger().WithName("setup")
+	deploy, err := getSelfDeploy(kcs, conf, setupLog)
 	if err != nil {
 		return nil, err
 	}
-	log.V(2).Info("using namespace: " + conf.NS)
+	setupLog.V(2).Info("using namespace: " + conf.NS)
+
+	if err := loadCRDs(cl, conf, setupLog); err != nil {
+		return nil, err
```

Comment:
can we log this error here or at least in NewManager

Reply:
https://github.com/Azure/aks-app-routing-operator/blob/d6fd1749935482477a7b2f88ab76dafdac750b6f/main.go#L24

Reply:
same way all errors in NewManager are shown